host/xtest/regression_1000.c

```diff
@@ -1473,3 +1476,266 @@ static void xtest_tee_test_1020(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(regression, 1020, xtest_tee_test_1020,
 		"Test lockdep algorithm");
+
+struct xtest_session {
+	ADBG_Case_t *c;
+	TEEC_Session session;
+	TEEC_Context context;
+};
+
+static TEEC_Result open_nsec_session(struct xtest_session *cs,
```

Comment:
This function looks correct, but may be a bit unnecessary. In other test cases we rely on `xtest_teec_ctx_init()` to already have been called to provide a global `TEEC_Context` and then we're using `xtest_teec_open_session()` as a helper function to open a session, much like this function.

So this function and `close_nsec_session()` could be replaced by calls to `xtest_teec_open_session()` and `TEEC_CloseSession()`.

Reply:
I got your point and I'm aware of these functions since 1012 sub tests are using it. Initially, when I took in consideration the usage of the funcs `xtest_teec_ctx_init()`, `xtest_teec_open_session()` and `xtest_teec_ctx_deinit()`, I realize that all of them are using the same global context represented by `xtest_teec_ctx`, but my code is using multiple contexts for different sub tests.

Do you think that we should use a single context? Maybe something is missing in my approach.

Reply:
I can't see anything in this test case needing or taking advantage of using different contexts. So unless there's a good reason not to use `xtest_teec_open_session()` etc I'd suggest do so.

Reply:
Ping?

Reply:
Sorry, I missed that. In the shortest time I will update the tests.

Reply:
The update makes use of `xtest_teec_open_session()` instead of `open_nsec_session()`. Kindly ask you to take a look over.